File: reference/SmartPharmacy-Kotlin/app/src/main/java/com/example/ui/screens/PurchaseScreens.kt
package com.example.ui.screens

import android.widget.Toast
import androidx.compose.foundation.background
import androidx.compose.foundation.border
import androidx.compose.foundation.clickable
import androidx.compose.foundation.layout.*
import androidx.compose.foundation.lazy.LazyColumn
import androidx.compose.foundation.lazy.items
import androidx.compose.foundation.rememberScrollState
import androidx.compose.foundation.shape.CircleShape
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.foundation.text.KeyboardOptions
import androidx.compose.foundation.verticalScroll
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.filled.*
import androidx.compose.material3.*
import androidx.compose.runtime.*
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.clip
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.platform.LocalContext
import androidx.compose.ui.platform.testTag
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.text.input.KeyboardType
import androidx.compose.ui.text.style.TextOverflow
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import com.example.data.model.PurchaseRecord
import com.example.ui.components.CustomButton
import com.example.ui.components.CustomSearchBar
import com.example.ui.components.CustomTextField
import com.example.ui.theme.*
import com.example.ui.viewmodel.PharmacyViewModel

@OptIn(ExperimentalMaterial3Api::class)
@Composable
fun PurchaseListScreen(
    viewModel: PharmacyViewModel,
    onNavigateToAddPurchase: () -> Unit,
    onNavigateToPurchaseDetails: (String) -> Unit
) {
    val purchases by viewModel.purchases.collectAsState()
    val searchQuery by viewModel.purchaseSearch.collectAsState()

    val filteredPurchases = remember(purchases, searchQuery) {
        purchases.filter {
            it.invoiceNumber.contains(searchQuery, ignoreCase = true) ||
                    it.distributorName.contains(searchQuery, ignoreCase = true) ||
                    it.medicineName.contains(searchQuery, ignoreCase = true)
        }
    }

    Scaffold(
        floatingActionButton = {
            FloatingActionButton(
                onClick = onNavigateToAddPurchase,
                containerColor = SecondaryTeal,
                contentColor = SurfaceWhite,
                shape = RoundedCornerShape(16.dp),
                modifier = Modifier.testTag("fab_add_purchase")
            ) {
                Row(
                    modifier = Modifier.padding(horizontal = 16.dp),
                    verticalAlignment = Alignment.CenterVertically
                ) {
                    Icon(imageVector = Icons.Default.Add, contentDescription = "Add Purchase")
                    Spacer(modifier = Modifier.width(6.dp))
                    Text("Add Purchase", fontWeight = FontWeight.Bold)
                }
            }
        },
        containerColor = SurfaceWhite
    ) { innerPadding ->
        Column(
            modifier = Modifier
                .fillMaxSize()
                .padding(innerPadding)
                .padding(horizontal = 16.dp)
        ) {
            Spacer(modifier = Modifier.height(12.dp))

            CustomSearchBar(
                query = searchQuery,
                onQueryChange = { viewModel.setPurchaseSearch(it) },
                placeholder = "Search invoice, distributor, medicine..."
            )

            Spacer(modifier = Modifier.height(12.dp))

            Text(
                text = "Total Purchases (${filteredPurchases.size})",
                fontSize = 14.sp,
                fontWeight = FontWeight.Bold,
                color = TextPrimary
            )

            Spacer(modifier = Modifier.height(8.dp))

            LazyColumn(
                verticalArrangement = Arrangement.spacedBy(10.dp),
                contentPadding = PaddingValues(bottom = 80.dp)
            ) {
                items(filteredPurchases, key = { it.id }) { purchase ->
                    Card(
                        onClick = { onNavigateToPurchaseDetails(purchase.id) },
                        shape = RoundedCornerShape(14.dp),
                        colors = CardDefaults.cardColors(containerColor = SurfaceWhite),
                        elevation = CardDefaults.cardElevation(defaultElevation = 1.dp),
                        modifier = Modifier
                            .fillMaxWidth()
                            .border(1.dp, SurfaceBorder, RoundedCornerShape(14.dp))
                    ) {
                        Row(
                            modifier = Modifier
                                .fillMaxWidth()
                                .padding(14.dp),
                            verticalAlignment = Alignment.CenterVertically
                        ) {
                            Box(
                                modifier = Modifier
                                    .size(44.dp)
                                    .clip(CircleShape)
                                    .background(SecondaryTealLight),
                                contentAlignment = Alignment.Center
                            ) {
                                Icon(
                                    imageVector = Icons.Default.ShoppingBag,
                                    contentDescription = null,
                                    tint = SecondaryTeal,
                                    modifier = Modifier.size(22.dp)
                                )
                            }

                            Spacer(modifier = Modifier.width(12.dp))

                            Column(modifier = Modifier.weight(1f)) {
                                Text(
                                    text = purchase.medicineName,
                                    fontSize = 15.sp,
                                    fontWeight = FontWeight.Bold,
                                    color = TextPrimary,
                                    maxLines = 1,
                                    overflow = TextOverflow.Ellipsis
                                )
                                Text(
                                    text = "${purchase.distributorName} • ${purchase.invoiceNumber}",
                                    fontSize = 12.sp,
                                    color = TextSecondary,
                                    maxLines = 1,
                                    overflow = TextOverflow.Ellipsis
                                )
                                Text(
                                    text = "Date: ${purchase.purchaseDate} • Qty: ${purchase.quantity}",
                                    fontSize = 11.sp,
                                    color = TextMuted
                                )
                            }

                            Spacer(modifier = Modifier.width(8.dp))

                            Text(
                                text = "₹${purchase.totalAmount.toInt()}",
                                fontSize = 15.sp,
                                fontWeight = FontWeight.Bold,
                                color = SecondaryTeal
                            )
                        }
                    }
                }
            }
        }
    }
}

@OptIn(ExperimentalMaterial3Api::class)
@Composable
fun AddPurchaseScreen(
    viewModel: PharmacyViewModel,
    onNavigateBack: () -> Unit
) {
    val context = LocalContext.current
    val distributors by viewModel.distributors.collectAsState()
    val medicines by viewModel.medicines.collectAsState()

    var selectedDistributor by remember { mutableStateOf(distributors.firstOrNull()?.name ?: "MedLife Wholesale") }
    var selectedMedicine by remember { mutableStateOf(medicines.firstOrNull()?.name ?: "Amoxicillin 500mg") }
    var invoiceNumber by remember { mutableStateOf("INV-PUR-2026-" + (1000..9999).random()) }
    var purchaseDate by remember { mutableStateOf("2026-07-25") }
    var quantityStr by remember { mutableStateOf("50") }
    var purchasePriceStr by remember { mutableStateOf("35.0") }
    var expiryDate by remember { mutableStateOf("2028-06-30") }

    var distExpanded by remember { mutableStateOf(false) }
    var medExpanded by remember { mutableStateOf(false) }

    Scaffold(
        topBar = {
            TopAppBar(
                title = { Text("New Purchase Order", fontWeight = FontWeight.Bold) },
                navigationIcon = {
                    IconButton(onClick = onNavigateBack) {
                        Icon(imageVector = Icons.Default.ArrowBack, contentDescription = "Back")
                    }
                },
                colors = TopAppBarDefaults.topAppBarColors(containerColor = SurfaceWhite)
            )
        },
        containerColor = SurfaceWhite
    ) { innerPadding ->
        Column(
            modifier = Modifier
                .fillMaxSize()
                .padding(innerPadding)
                .padding(16.dp)
                .verticalScroll(rememberScrollState()),
            verticalArrangement = Arrangement.spacedBy(14.dp)
        ) {
            // Distributor Selection
            ExposedDropdownMenuBox(
                expanded = distExpanded,
                onExpandedChange = { distExpanded = !distExpanded }
            ) {
                CustomTextField(
                    value = selectedDistributor,
                    onValueChange = {},
                    label = "Distributor / Supplier",
                    leadingIcon = Icons.Default.LocalShipping,
                    trailingIcon = { ExposedDropdownMenuDefaults.TrailingIcon(expanded = distExpanded) },
                    modifier = Modifier.menuAnchor()
                )

                ExposedDropdownMenu(
                    expanded = distExpanded,
                    onDismissRequest = { distExpanded = false },
                    modifier = Modifier.background(SurfaceWhite)
                ) {
                    distributors.forEach { dist ->
                        DropdownMenuItem(
                            text = { Text(dist.name) },
                            onClick = {
                                selectedDistributor = dist.name
                                distExpanded = false
                            }
                        )
                    }
                }
            }

            CustomTextField(
                value = invoiceNumber,
                onValueChange = { invoiceNumber = it },
                label = "Invoice Number",
                leadingIcon = Icons.Default.Receipt
            )

            // Medicine Selection
            ExposedDropdownMenuBox(
                expanded = medExpanded,
                onExpandedChange = { medExpanded = !medExpanded }
            ) {
                CustomTextField(
                    value = selectedMedicine,
                    onValueChange = {},
                    label = "Select Medicine",
                    leadingIcon = Icons.Default.Medication,
                    trailingIcon = { ExposedDropdownMenuDefaults.TrailingIcon(expanded = medExpanded) },
                    modifier = Modifier.menuAnchor()
                )

                ExposedDropdownMenu(
                    expanded = medExpanded,
                    onDismissRequest = { medExpanded = false },
                    modifier = Modifier.background(SurfaceWhite)
                ) {
                    medicines.forEach { med ->
                        DropdownMenuItem(
                            text = { Text("${med.name} (${med.company})") },
                            onClick = {
                                selectedMedicine = med.name
                                purchasePriceStr = med.purchasePrice.toString()
                                medExpanded = false
                            }
                        )
                    }
                }
            }

            Row(
                modifier = Modifier.fillMaxWidth(),
                horizontalArrangement = Arrangement.spacedBy(10.dp)
            ) {
                CustomTextField(
                    value = quantityStr,
                    onValueChange = { quantityStr = it },
                    label = "Quantity Purchased",
                    placeholder = "50",
                    leadingIcon = Icons.Default.FormatListNumbered,
                    keyboardOptions = KeyboardOptions(keyboardType = KeyboardType.Number),
                    modifier = Modifier.weight(1f)
                )

                CustomTextField(
                    value = purchasePriceStr,
                    onValueChange = { purchasePriceStr = it },
                    label = "Purchase Price (₹)",
                    placeholder = "35.0",
                    leadingIcon = Icons.Default.AttachMoney,
                    keyboardOptions = KeyboardOptions(keyboardType = KeyboardType.Number),
                    modifier = Modifier.weight(1f)
                )
            }

            Row(
                modifier = Modifier.fillMaxWidth(),
                horizontalArrangement = Arrangement.spacedBy(10.dp)
            ) {
                CustomTextField(
                    value = purchaseDate,
                    onValueChange = { purchaseDate = it },
                    label = "Purchase Date",
                    leadingIcon = Icons.Default.CalendarToday,
                    modifier = Modifier.weight(1f)
                )

                CustomTextField(
                    value = expiryDate,
                    onValueChange = { expiryDate = it },
                    label = "Batch Expiry",
                    leadingIcon = Icons.Default.Event,
                    modifier = Modifier.weight(1f)
                )
            }

            val qty = quantityStr.toIntOrNull() ?: 0
            val price = purchasePriceStr.toDoubleOrNull() ?: 0.0
            val total = qty * price

            Card(
                shape = RoundedCornerShape(12.dp),
                colors = CardDefaults.cardColors(containerColor = SecondaryTealLight),
                modifier = Modifier.fillMaxWidth()
            ) {
                Row(
                    modifier = Modifier.padding(16.dp),
                    horizontalArrangement = Arrangement.SpaceBetween,
                    verticalAlignment = Alignment.CenterVertically
                ) {
                    Text("Total Invoice Amount:", fontWeight = FontWeight.Bold, fontSize = 14.sp)
                    Text("₹${String.format("%.2f", total)}", fontWeight = FontWeight.Bold, fontSize = 18.sp, color = SecondaryTeal)
                }
            }

            Spacer(modifier = Modifier.height(16.dp))

            CustomButton(
                text = "Save Purchase & Update Stock",
                containerColor = SecondaryTeal,
                onClick = {
                    val targetMed = medicines.find { it.name.equals(selectedMedicine, ignoreCase = true) }
                    val medId = targetMed?.id ?: "M01"

                    val record = PurchaseRecord(
                        id = "P" + System.currentTimeMillis().toString().takeLast(4),
                        invoiceNumber = invoiceNumber,
                        distributorName = selectedDistributor,
                        purchaseDate = purchaseDate,
                        medicineId = medId,
                        medicineName = selectedMedicine,
                        quantity = qty,
                        purchasePrice = price,
                        expiryDate = expiryDate,
                        totalAmount = total
                    )

                    viewModel.addPurchase(record)
                    Toast.makeText(context, "Purchase saved! Medicine stock increased by $qty", Toast.LENGTH_LONG).show()
                    onNavigateBack()
                }
            )
        }
    }
}

@OptIn(ExperimentalMaterial3Api::class)
@Composable
fun PurchaseDetailsScreen(
    viewModel: PharmacyViewModel,
    purchaseId: String,
    onNavigateBack: () -> Unit
) {
    val purchases by viewModel.purchases.collectAsState()
    val purchase = purchases.find { it.id == purchaseId }

    if (purchase == null) {
        Box(modifier = Modifier.fillMaxSize(), contentAlignment = Alignment.Center) {
            Text("Purchase record not found")
        }
        return
    }

    Scaffold(
        topBar = {
            TopAppBar(
                title = { Text("Purchase Details", fontWeight = FontWeight.Bold) },
                navigationIcon = {
                    IconButton(onClick = onNavigateBack) {
                        Icon(imageVector = Icons.Default.ArrowBack, contentDescription = "Back")
                    }
                },
                colors = TopAppBarDefaults.topAppBarColors(containerColor = SurfaceWhite)
            )
        },
        containerColor = SurfaceWhite
    ) { innerPadding ->
        Column(
            modifier = Modifier
                .fillMaxSize()
                .padding(innerPadding)
                .padding(16.dp),
            verticalArrangement = Arrangement.spacedBy(16.dp)
        ) {
            Card(
                shape = RoundedCornerShape(16.dp),
                colors = CardDefaults.cardColors(containerColor = SecondaryTealLight),
                modifier = Modifier.fillMaxWidth()
            ) {
                Column(modifier = Modifier.padding(20.dp)) {
                    Text(text = "Invoice: ${purchase.invoiceNumber}", fontSize = 18.sp, fontWeight = FontWeight.Bold, color = SecondaryTeal)
                    Spacer(modifier = Modifier.height(4.dp))
                    Text(text = "Supplier: ${purchase.distributorName}", fontSize = 14.sp, color = TextPrimary)
                }
            }

            Card(
                shape = RoundedCornerShape(14.dp),
                colors = CardDefaults.cardColors(containerColor = SurfaceWhite),
                elevation = CardDefaults.cardElevation(defaultElevation = 1.dp),
                modifier = Modifier
                    .fillMaxWidth()
                    .border(1.dp, SurfaceBorder, RoundedCornerShape(14.dp))
            ) {
                Column(
                    modifier = Modifier.padding(18.dp),
                    verticalArrangement = Arrangement.spacedBy(12.dp)
                ) {
                    DetailItem(label = "Medicine Name", value = purchase.medicineName)
                    DetailItem(label = "Purchase Date", value = purchase.purchaseDate)
                    DetailItem(label = "Quantity Received", value = "${purchase.quantity} units")
                    DetailItem(label = "Unit Purchase Price", value = "₹${purchase.purchasePrice}")
                    DetailItem(label = "Batch Expiry Date", value = purchase.expiryDate)
                    HorizontalDivider(color = SurfaceBorder)
                    DetailItem(label = "Grand Total Amount", value = "₹${purchase.totalAmount}", isBold = true, valueColor = SecondaryTeal)
                }
            }
        }
    }
}

@Composable
private fun DetailItem(
    label: String,
    value: String,
    isBold: Boolean = false,
    valueColor: Color = TextPrimary
) {
    Row(
        modifier = Modifier.fillMaxWidth(),
        horizontalArrangement = Arrangement.SpaceBetween
    ) {
        Text(text = label, fontSize = 13.sp, color = TextSecondary)
        Text(
            text = value,
            fontSize = 14.sp,
            fontWeight = if (isBold) FontWeight.Bold else FontWeight.SemiBold,
            color = valueColor
        )
    }
}

File: lib/screens/purchase_screens.dart
import 'package:flutter/material.dart';
import 'package:provider/provider.dart';
import '../core/theme/app_colors.dart';
import '../models/purchase_record.dart';
import '../providers/pharmacy_provider.dart';
import '../routes/app_routes.dart';
import '../widgets/common/custom_button.dart';
import '../widgets/common/custom_search_bar.dart';
import '../widgets/common/custom_text_field.dart';
import '../widgets/common/purchase_card.dart';

class PurchaseListScreen extends StatefulWidget {
  const PurchaseListScreen({super.key});

  @override
  State<PurchaseListScreen> createState() => _PurchaseListScreenState();
}

class _PurchaseListScreenState extends State<PurchaseListScreen> {
  late TextEditingController _searchController;

  @override
  void initState() {
    super.initState();
    final provider = context.read<PharmacyProvider>();
    _searchController = TextEditingController(text: provider.purchaseSearch);
  }

  @override
  void dispose() {
    _searchController.dispose();
    super.dispose();
  }

  @override
  Widget build(BuildContext context) {
    final provider = context.watch<PharmacyProvider>();
    final purchases = provider.purchases;
    final searchQuery = provider.purchaseSearch;

    final filteredPurchases = purchases.where((p) {
      return p.invoiceNumber.toLowerCase().contains(searchQuery.toLowerCase()) ||
          p.distributorName.toLowerCase().contains(searchQuery.toLowerCase()) ||
          p.medicineName.toLowerCase().contains(searchQuery.toLowerCase());
    }).toList();

    return Scaffold(
      backgroundColor: AppColors.surfaceWhite,
      appBar: AppBar(
        backgroundColor: AppColors.surfaceWhite,
        elevation: 0,
        title: const Text(
          "Purchase History",
          style: TextStyle(fontWeight: FontWeight.bold, color: AppColors.textPrimary),
        ),
        leading: IconButton(
          icon: const Icon(Icons.arrow_back, color: AppColors.textPrimary),
          onPressed: () => Navigator.pop(context),
        ),
      ),
      floatingActionButton: FloatingActionButton.extended(
        onPressed: () => Navigator.pushNamed(context, AppRoutes.addPurchase),
        backgroundColor: AppColors.secondaryTeal,
        foregroundColor: AppColors.surfaceWhite,
        shape: RoundedRectangleBorder(borderRadius: BorderRadius.circular(16)),
        icon: const Icon(Icons.add),
        label: const Text(
          "Add Purchase",
          style: TextStyle(fontWeight: FontWeight.bold),
        ),
      ),
      body: SafeArea(
        child: Column(
          crossAxisAlignment: CrossAxisAlignment.start,
          children: [
            const SizedBox(height: 12),
            Padding(
              padding: const EdgeInsets.symmetric(horizontal: 16.0),
              child: CustomSearchBar(
                controller: _searchController,
                hint: "Search invoice, distributor, medicine...",
                onChanged: (val) => provider.setPurchaseSearch(val),
              ),
            ),
            const SizedBox(height: 12),
            Padding(
              padding: const EdgeInsets.symmetric(horizontal: 16.0),
              child: Text(
                "Total Purchases (${filteredPurchases.length})",
                style: const TextStyle(
                  fontSize: 14,
                  fontWeight: FontWeight.bold,
                  color: AppColors.textPrimary,
                ),
              ),
            ),
            const SizedBox(height: 8),
            Expanded(
              child: filteredPurchases.isEmpty
                  ? _buildEmptyState()
                  : ListView.separated(
                      padding: const EdgeInsets.fromLTRB(16, 0, 16, 80),
                      itemCount: filteredPurchases.length,
                      separatorBuilder: (context, index) => const SizedBox(height: 10),
                      itemBuilder: (context, index) {
                        final purchase = filteredPurchases[index];
                        return PurchaseCard(
                          purchase: purchase,
                          onTap: () => Navigator.pushNamed(
                            context,
                            AppRoutes.purchaseDetails,
                            arguments: purchase.id,
                          ),
                        );
                      },
                    ),
            ),
          ],
        ),
      ),
    );
  }

  Widget _buildEmptyState() {
    return Center(
      child: Column(
        mainAxisAlignment: MainAxisAlignment.center,
        children: const [
          Icon(Icons.shopping_bag_outlined, size: 48, color: AppColors.textMuted),
          SizedBox(height: 12),
          Text(
            "No purchase records found",
            style: TextStyle(fontSize: 16, fontWeight: FontWeight.bold, color: AppColors.textPrimary),
          ),
          Text(
            "Try adjusting your search query",
            style: TextStyle(fontSize: 13, color: AppColors.textSecondary),
          ),
        ],
      ),
    );
  }
}

class AddPurchaseScreen extends StatefulWidget {
  const AddPurchaseScreen({super.key});

  @override
  State<AddPurchaseScreen> createState() => _AddPurchaseScreenState();
}

class _AddPurchaseScreenState extends State<AddPurchaseScreen> {
  final _invoiceController = TextEditingController();
  final _quantityController = TextEditingController();
  final _priceController = TextEditingController();
  final _purchaseDateController = TextEditingController();
  final _expiryDateController = TextEditingController();

  String _selectedDistributor = "";
  String _selectedMedicine = "";

  @override
  void initState() {
    super.initState();
    final provider = context.read<PharmacyProvider>();
    
    if (provider.distributors.isNotEmpty) {
      _selectedDistributor = provider.distributors.first.name;
    } else {
      _selectedDistributor = "MedLife Wholesale";
    }

    if (provider.medicines.isNotEmpty) {
      _selectedMedicine = provider.medicines.first.name;
      _priceController.text = provider.medicines.first.purchasePrice.toString();
    } else {
      _selectedMedicine = "Amoxicillin 500mg";
      _priceController.text = "35.0";
    }

    _invoiceController.text = "INV-PUR-2026-${1000 + (DateTime.now().millisecond % 9000)}";
    _purchaseDateController.text = "2026-07-25";
    _quantityController.text = "50";
    _expiryDateController.text = "2028-06-30";
  }

  @override
  void dispose() {
    _invoiceController.dispose();
    _quantityController.dispose();
    _priceController.dispose();
    _purchaseDateController.dispose();
    _expiryDateController.dispose();
    super.dispose();
  }

  void _savePurchase() {
    final qty = int.tryParse(_quantityController.text) ?? 0;
    final price = double.tryParse(_priceController.text) ?? 0.0;
    final total = qty * price;

    final provider = context.read<PharmacyProvider>();
    final targetMed = provider.medicines.firstWhere(
      (m) => m.name.toLowerCase() == _selectedMedicine.toLowerCase(),
      orElse: () => provider.medicines.first,
    );

    final record = PurchaseRecord(
      id: "P${DateTime.now().millisecondsSinceEpoch.toString().substring(DateTime.now().millisecondsSinceEpoch.toString().length - 4)}",
      invoiceNumber: _invoiceController.text.trim(),
      distributorName: _selectedDistributor,
      purchaseDate: _purchaseDateController.text.trim(),
      medicineId: targetMed.id,
      medicineName: _selectedMedicine,
      quantity: qty,
      purchasePrice: price,
      expiryDate: _expiryDateController.text.trim(),
      totalAmount: total,
    );

    provider.addPurchase(record);
    ScaffoldMessenger.of(context).showSnackBar(
      SnackBar(content: Text("Purchase saved! Medicine stock increased by $qty")),
    );
    Navigator.pop(context);
  }

  @override
  Widget build(BuildContext context) {
    final provider = context.watch<PharmacyProvider>();
    final qty = int.tryParse(_quantityController.text) ?? 0;
    final price = double.tryParse(_priceController.text) ?? 0.0;
    final total = qty * price;

    return Scaffold(
      backgroundColor: AppColors.surfaceWhite,
      appBar: AppBar(
        backgroundColor: AppColors.surfaceWhite,
        elevation: 0,
        title: const Text(
          "New Purchase Order",
          style: TextStyle(fontWeight: FontWeight.bold, color: AppColors.textPrimary),
        ),
        leading: IconButton(
          icon: const Icon(Icons.arrow_back, color: AppColors.textPrimary),
          onPressed: () => Navigator.pop(context),
        ),
      ),
      body: SingleChildScrollView(
        padding: const EdgeInsets.all(16.0),
        child: Column(
          children: [
            _buildDropdownField(
              label: "Distributor / Supplier",
              value: _selectedDistributor,
              icon: Icons.local_shipping,
              items: provider.distributors.map((d) => d.name).toList(),
              onChanged: (val) {
                setState(() {
                  _selectedDistributor = val!;
                });
              },
            ),
            const SizedBox(height: 14),
            CustomTextField(
              controller: _invoiceController,
              label: "Invoice Number",
              leadingIcon: Icons.receipt,
            ),
            const SizedBox(height: 14),
            _buildDropdownField(
              label: "Select Medicine",
              value: _selectedMedicine,
              icon: Icons.medication,
              items: provider.medicines.map((m) => m.name).toList(),
              onChanged: (val) {
                setState(() {
                  _selectedMedicine = val!;
                  final med = provider.medicines.firstWhere((m) => m.name == val);
                  _priceController.text = med.purchasePrice.toString();
                });
              },
            ),
            const SizedBox(height: 14),
            Row(
              children: [
                Expanded(
                  child: CustomTextField(
                    controller: _quantityController,
                    label: "Quantity Purchased",
                    hint: "50",
                    leadingIcon: Icons.format_list_numbered,
                    keyboardType: TextInputType.number,
                    onChanged: (_) => setState(() {}),
                  ),
                ),
                const SizedBox(width: 10),
                Expanded(
                  child: CustomTextField(
                    controller: _priceController,
                    label: "Purchase Price (₹)",
                    hint: "35.0",
                    leadingIcon: Icons.attach_money,
                    keyboardType: TextInputType.number,
                    onChanged: (_) => setState(() {}),
                  ),
                ),
              ],
            ),
            const SizedBox(height: 14),
            Row(
              children: [
                Expanded(
                  child: CustomTextField(
                    controller: _purchaseDateController,
                    label: "Purchase Date",
                    leadingIcon: Icons.calendar_today,
                  ),
                ),
                const SizedBox(width: 10),
                Expanded(
                  child: CustomTextField(
                    controller: _expiryDateController,
                    label: "Batch Expiry",
                    leadingIcon: Icons.event,
                  ),
                ),
              ],
            ),
            const SizedBox(height: 20),
            Card(
              elevation: 0,
              margin: EdgeInsets.zero,
              shape: RoundedRectangleBorder(borderRadius: BorderRadius.circular(12)),
              color: AppColors.secondaryTealLight,
              child: Padding(
                padding: const EdgeInsets.all(16.0),
                child: Row(
                  mainAxisAlignment: MainAxisAlignment.spaceBetween,
                  children: [
                    const Text(
                      "Total Invoice Amount:",
                      style: TextStyle(fontWeight: FontWeight.bold, fontSize: 14),
                    ),
                    Text(
                      "₹${total.toStringAsFixed(2)}",
                      style: const TextStyle(
                        fontWeight: FontWeight.bold,
                        fontSize: 18,
                        color: AppColors.secondaryTeal,
                      ),
                    ),
                  ],
                ),
              ),
            ),
            const SizedBox(height: 24),
            CustomButton(
              text: "Save Purchase & Update Stock",
              containerColor: AppColors.secondaryTeal,
              onPressed: _savePurchase,
            ),
          ],
        ),
      ),
    );
  }

  Widget _buildDropdownField({
    required String label,
    required String value,
    required IconData icon,
    required List<String> items,
    required void Function(String?) onChanged,
  }) {
    String? effectiveValue = items.contains(value) ? value : (items.isNotEmpty ? items.first : null);

    return DropdownButtonFormField<String>(
      initialValue: effectiveValue,
      decoration: InputDecoration(
        labelText: label,
        prefixIcon: Icon(icon, color: AppColors.primaryBlue),
        filled: true,
        fillColor: AppColors.surfaceWhite,
        contentPadding: const EdgeInsets.symmetric(horizontal: 16, vertical: 16),
        border: OutlineInputBorder(
          borderRadius: BorderRadius.circular(12),
          borderSide: const BorderSide(color: AppColors.surfaceBorder),
        ),
        enabledBorder: OutlineInputBorder(
          borderRadius: BorderRadius.circular(12),
          borderSide: const BorderSide(color: AppColors.surfaceBorder),
        ),
        focusedBorder: OutlineInputBorder(
          borderRadius: BorderRadius.circular(12),
          borderSide: const BorderSide(color: AppColors.primaryBlue),
        ),
        labelStyle: const TextStyle(color: AppColors.textSecondary),
      ),
      items: items.map((String val) {
        return DropdownMenuItem<String>(
          value: val,
          child: Text(
            val,
            style: const TextStyle(color: AppColors.textPrimary),
            overflow: TextOverflow.ellipsis,
          ),
        );
      }).toList(),
      onChanged: onChanged,
      icon: const Icon(Icons.arrow_drop_down, color: AppColors.textSecondary),
      dropdownColor: AppColors.surfaceWhite,
      isExpanded: true,
    );
  }
}

class PurchaseDetailsScreen extends StatelessWidget {
  final String purchaseId;

  const PurchaseDetailsScreen({
    super.key,
    required this.purchaseId,
  });

  @override
  Widget build(BuildContext context) {
    final provider = context.watch<PharmacyProvider>();
    final purchase = provider.purchases.firstWhere(
      (p) => p.id == purchaseId,
      orElse: () => throw Exception("Purchase record not found"),
    );

    return Scaffold(
      backgroundColor: AppColors.surfaceWhite,
      appBar: AppBar(
        backgroundColor: AppColors.surfaceWhite,
        elevation: 0,
        title: const Text(
          "Purchase Details",
          style: TextStyle(fontWeight: FontWeight.bold, color: AppColors.textPrimary),
        ),
        leading: IconButton(
          icon: const Icon(Icons.arrow_back, color: AppColors.textPrimary),
          onPressed: () => Navigator.pop(context),
        ),
      ),
      body: Padding(
        padding: const EdgeInsets.all(16.0),
        child: Column(
          children: [
            Card(
              elevation: 0,
              margin: EdgeInsets.zero,
              shape: RoundedRectangleBorder(borderRadius: BorderRadius.circular(16)),
              color: AppColors.secondaryTealLight,
              child: Padding(
                padding: const EdgeInsets.all(20.0),
                child: Column(
                  crossAxisAlignment: CrossAxisAlignment.start,
                  children: [
                    Text(
                      "Invoice: ${purchase.invoiceNumber}",
                      style: const TextStyle(
                        fontSize: 18,
                        fontWeight: FontWeight.bold,
                        color: AppColors.secondaryTeal,
                      ),
                    ),
                    const SizedBox(height: 4),
                    Text(
                      "Supplier: ${purchase.distributorName}",
                      style: const TextStyle(fontSize: 14, color: AppColors.textPrimary),
                    ),
                  ],
                ),
              ),
            ),
            const SizedBox(height: 16),
            Card(
              elevation: 1,
              margin: EdgeInsets.zero,
              shape: RoundedRectangleBorder(
                borderRadius: BorderRadius.circular(14),
                side: const BorderSide(color: AppColors.surfaceBorder),
              ),
              color: AppColors.surfaceWhite,
              child: Padding(
                padding: const EdgeInsets.all(18.0),
                child: Column(
                  children: [
                    _buildDetailItem("Medicine Name", purchase.medicineName),
                    const SizedBox(height: 12),
                    _buildDetailItem("Purchase Date", purchase.purchaseDate),
                    const SizedBox(height: 12),
                    _buildDetailItem("Quantity Received", "${purchase.quantity} units"),
                    const SizedBox(height: 12),
                    _buildDetailItem("Unit Purchase Price", "₹${purchase.purchasePrice}"),
                    const SizedBox(height: 12),
                    _buildDetailItem("Batch Expiry Date", purchase.expiryDate),
                    const Divider(height: 24, color: AppColors.surfaceBorder),
                    _buildDetailItem(
                      "Grand Total Amount",
                      "₹${purchase.totalAmount}",
                      isBold: true,
                      valueColor: AppColors.secondaryTeal,
                    ),
                  ],
                ),
              ),
            ),
          ],
        ),
      ),
    );
  }

  Widget _buildDetailItem(
    String label,
    String value, {
    bool isBold = false,
    Color valueColor = AppColors.textPrimary,
  }) {
    return Row(
      mainAxisAlignment: MainAxisAlignment.spaceBetween,
      children: [
        Text(
          label,
          style: const TextStyle(fontSize: 13, color: AppColors.textSecondary),
        ),
        Text(
          value,
          style: TextStyle(
            fontSize: 14,
            fontWeight: isBold ? FontWeight.bold : FontWeight.w600,
            color: valueColor,
          ),
        ),
      ],
    );
  }
}
